src/decode.rs

```diff
@@ -3525,280 +3525,231 @@ unsafe fn decode_b(
+                        b.cfl_alpha()[1],
                         ts.msac.rng,
                     );
                 }
             } else if b_dim[2] + b_dim[3] >= 2
-                && b.c2rust_unnamed.c2rust_unnamed.uv_mode as libc::c_int
-                    >= VERT_PRED as libc::c_int
-                && b.c2rust_unnamed.c2rust_unnamed.uv_mode as libc::c_int
-                    <= VERT_LEFT_PRED as libc::c_int
+                && b.uv_mode() >= VERT_PRED as u8
+                && b.uv_mode() <= VERT_LEFT_PRED as u8
             {
-                let acdf_0: *mut uint16_t =
-                    (ts.cdf.m.angle_delta[(b.c2rust_unnamed.c2rust_unnamed.uv_mode as libc::c_int
-                        - VERT_PRED as libc::c_int)
-                        as usize])
-                        .as_mut_ptr();
-                let angle_0 = dav1d_msac_decode_symbol_adapt8(
-                    &mut ts.msac,
-                    acdf_0,
-                    6 as libc::c_int as size_t,
-                ) as libc::c_int;
-                b.c2rust_unnamed.c2rust_unnamed.uv_angle = (angle_0 - 3) as int8_t;
+                let acdf = &mut ts.cdf.m.angle_delta[b.uv_mode() as usize - VERT_PRED as usize];
+                let angle = dav1d_msac_decode_symbol_adapt8(&mut ts.msac, acdf.as_mut_ptr(), 6)
+                    as libc::c_int;
+                *b.uv_angle_mut() = (angle - 3) as int8_t;
             }
         }
-        b.c2rust_unnamed.c2rust_unnamed.pal_sz[1] = 0 as libc::c_int as uint8_t;
-        b.c2rust_unnamed.c2rust_unnamed.pal_sz[0] = b.c2rust_unnamed.c2rust_unnamed.pal_sz[1];
+
+        *b.pal_sz_mut() = [0, 0];
         if frame_hdr.allow_screen_content_tools != 0 && imax(bw4, bh4) <= 16 && bw4 + bh4 >= 4 {
-            let sz_ctx = (b_dim[2] + b_dim[3] - 2) as libc::c_int;
-            if b.c2rust_unnamed.c2rust_unnamed.y_mode as libc::c_int == DC_PRED as libc::c_int {
-                let pal_ctx = ((*t.a).pal_sz[bx4 as usize] as libc::c_int > 0) as libc::c_int
-                    + (t.l.pal_sz[by4 as usize] as libc::c_int > 0) as libc::c_int;
+            let sz_ctx = b_dim[2] + b_dim[3] - 2;
+            if b.y_mode() == DC_PRED as u8 {
+                let pal_ctx = ((*t.a).pal_sz[bx4 as usize] > 0) as usize
+                    + (t.l.pal_sz[by4 as usize] > 0) as usize;
                 let use_y_pal = dav1d_msac_decode_bool_adapt(
                     &mut ts.msac,
-                    (ts.cdf.m.pal_y[sz_ctx as usize][pal_ctx as usize]).as_mut_ptr(),
-                ) as libc::c_int;
+                    ts.cdf.m.pal_y[sz_ctx as usize][pal_ctx].as_mut_ptr(),
+                ) != 0;
+
                 if DEBUG_BLOCK_INFO(f, t) {
-                    printf(
-                        b"Post-y_pal[%d]: r=%d\n\0" as *const u8 as *const libc::c_char,
-                        use_y_pal,
-                        ts.msac.rng,
-                    );
+                    println!("Post-y_pal[{}]: r={}", use_y_pal, ts.msac.rng);
```

Comment:
`use_y_pal` is a `bool` now instead of `int`, so it'll print differently here.  That might be okay, but I want sure since it's printed inside of `[]` and I wasn't sure if it's supposed to be an index, in which case casting it to a number is better.

Reply:
I changed it deliberately, it doesn't look like `use_y_pal` is being used as an index (it's only used in an `if` block condition), so I figured changing the output here would make things more clear. I can change it back to printing as a number if you think that'd be clearer though.

Reply:
I think it's fine your way, as a `bool`.  I just want to make sure it's deliberate.  It's debug output, and if we need to change it back later for whatever reason, it should be simple.